server/sandbox_remove.go

```diff
@@ -67,6 +67,13 @@ func (s *Server) removePodSandbox(ctx context.Context, sb *sandbox.Sandbox) erro
 		return fmt.Errorf("failed to delete pod sandbox %s from index: %w", sb.ID(), err)
 	}
 
+	if s.config.EventedPLEG {
+		if err := s.Runtime().UpdateContainerStatus(ctx, sb.InfraContainer()); err != nil {
```

Comment:
why this UpdateContainerStatus?

Reply:
@haircommander without this we found that kubelet pod cache was not getting correctly updated. In Generic PLEG in about 1s the cache is refreshed, but in evented pleg there is no update to the cache unless another event is generated for that pod. During testing, we saw that kubelet was stuck with old pod status without that call to `UpdateContainerStatus`. 

Even though they are expensive, they are executed only in case the evented pleg is enabled.

Reply:
yeah but in the future evented pleg will be the default. Maybe we should figure out how to update the status more efficiently so we incur less overhead. I guess we don't have to block, but we should open an issue and add TODOs to remove them or make them more efficient

Reply:
also, would you mind reminding me, it's my naive assumption the PLEG causes the pod cache to update. Why is sending the event not sufficient?

Reply:
The way it works is, 
1. CRIO sends an event
2. Kubelet requests pod status 
3. Kube cache is updated with the latest pod status. 

During testing we had noticed that incorrect pod status was getting updated in the kubelet without `UpdateContainerStatus`. Having said that, I need to some testing to see if we can remove this call from `remove` operation.